narwhals/_dask/dataframe.py

```diff
@@ -79,7 +79,9 @@ def filter(
             and isinstance(predicates[0], list)
             and all(isinstance(x, bool) for x in predicates[0])
         ):
-            msg = "Filtering with boolean mask is not supported for `DaskLazyFrame`"
+            msg = (
+                "`LazyFrame.filter` is not supported for Dask backend with boolean masks."
```

Comment:
Aligns error message with others. User should not really know about `DaskLazyFrame`